hnn_core/viz.py

```diff
@@ -242,29 +242,36 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    from .dipole import Dipole
+    from .dipole import Dipole, average_dipoles
 
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
     if isinstance(dpl, Dipole):
-        dpl = [dpl]
+        dpls = [dpl]
+    else:
+        # add average plot only when we have  trials>1.
+        dpls = dpl + [average_dipoles(dpl)]
 
-    scale_applied = dpl[0].scale_applied
-    for dpl_trial in dpl:
+    scale_applied = dpls[0].scale_applied
+    for i, dpl_trial in enumerate(dpls):
```

Comment:
this is a nitpick, but we make a lot of those :) I would rename the variable `i` to be `idx` as short for "index". Single letter variable names are not recommended typically because they are a bit hard to Ctr+F (find) and replace.

Reply:
OK